include/swift/Parse/Lexer.h

```diff
@@ -153,6 +153,7 @@ class Lexer {
         CommentRetentionMode RetainComments);
 
   void initialize(unsigned Offset, unsigned EndOffset);
+  void initialize(Optional<CharSourceRange> Range);
```

Comment:
Could you add a comment that the lexer lexes the whole buffer if `Range` is `None`?